app/js/transformers/matches.js

```diff
@@ -0,0 +1,134 @@
+const mapMatchesToDates = ({ matches, teams }) => {
+    return matches.reduce((dates, match) => {
+        const matchDay = getDayFromDate(match.date);
+        const date = dates.find(date => date.day === matchDay);
+        
+        Object.assign(match.homeTeam, getTeamDetails(match.homeTeam.id, teams), getPlayersStats(match.homeTeam.players));
```

Comment:
I think you are right, did not spend too much time on these functions since they will be lambda-s eventually and need to be refactor anyways. Thanks for the comment!